Server/mods/deathmatch/logic/luadefs/CLuaTeamDefs.cpp

```diff
@@ -67,282 +65,86 @@ void CLuaTeamDefs::AddClass(lua_State* luaVM)
     lua_registerclass(luaVM, "Team", "Element");
 }
 
-int CLuaTeamDefs::SetPlayerTeam(lua_State* luaVM)
+std::variant<CTeam*, bool> CLuaTeamDefs::CreateTeam(lua_State* lua, const std::string_view name, const std::uint8_t red, const std::uint8_t green, const std::uint8_t blue) noexcept
 {
-    CPlayer* pPlayer;
-    CTeam*   pTeam;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pPlayer);
-    argStream.ReadUserData(pTeam, NULL);
+    CLuaMain& vm = lua_getownercluamain(lua);
+    CResource* resource = vm.GetResource();
 
-    if (!argStream.HasErrors())
-    {
-        if (CStaticFunctionDefinitions::SetPlayerTeam(pPlayer, pTeam))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    if (!resource)
+        return false;
 
-    lua_pushboolean(luaVM, false);
-    return 1;
-}
+    std::string string(name);
+    CTeam* team = CStaticFunctionDefinitions::CreateTeam(resource, string.c_str(), red, green, blue);
 
-int CLuaTeamDefs::CreateTeam(lua_State* luaVM)
-{
-    SString       strName;
-    unsigned char ucRed;
-    unsigned char ucGreen;
-    unsigned char ucBlue;
+    if (!team)
+        return false;
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadString(strName);
-    argStream.ReadNumber(ucRed, 235);
-    argStream.ReadNumber(ucGreen, 221);
-    argStream.ReadNumber(ucBlue, 178);
+    CElementGroup* group = resource->GetElementGroup();
 
-    if (!argStream.HasErrors())
-    {
-        CLuaMain* pLuaMain = g_pGame->GetLuaManager()->GetVirtualMachine(luaVM);
-        if (pLuaMain)
-        {
-            CResource* pResource = pLuaMain->GetResource();
-            if (pResource)
-            {
-                CTeam* pTeam = CStaticFunctionDefinitions::CreateTeam(pResource, strName, ucRed, ucGreen, ucBlue);
-                if (pTeam)
-                {
-                    CElementGroup* pGroup = pResource->GetElementGroup();
-                    if (pGroup)
-                    {
-                        pGroup->Add(pTeam);
-                    }
-                    lua_pushelement(luaVM, pTeam);
-                    return 1;
-                }
-            }
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    if (group)
+        group->Add(team);
 
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return team;
 }
 
-int CLuaTeamDefs::GetPlayersInTeam(lua_State* luaVM)
+std::variant<CTeam*, bool> CLuaTeamDefs::GetTeamFromName(const std::string_view name) noexcept
 {
-    CTeam* pTeam;
+    std::string string(name);
```

Comment:
Remove the `noexcept` here, and change the parameter type from `std::string_view` to `std::string`.